src/app.js

```diff
@@ -23,6 +25,15 @@ class SAFEApp extends EventEmitter {
   */
   constructor(appInfo, networkStateCallBack) {
     super();
+
+    const appInfoMustHaveProperties = ['id', 'name', 'vendor', 'scope'];
+    const hasCorrectProperties = appInfoMustHaveProperties.every(
```

Comment:
Just validating has own property wont be enough. Must check whether the value is not empty.

Can add an function `isValid` to the AppInfo class. This function can validate the instance and return boolean. Would be better to validate it in the model itself than doing it in the constructor.
```
isValid() {
   const mandatoryFields = ['id', 'name', 'vendor', 'scope'];
   for (let i = 0; i < mandatoryFields.length; i++) {
     if (!(mandatoryFields[i] && mandatoryFields[i].trim())) {
        return false;
     }
   }
   return true;
}
```